src/test/java/uk/gov/homeoffice/digital/sas/person/CallistoPersonRestApiIntegrationTest.java

```diff
@@ -0,0 +1,182 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import java.io.UnsupportedEncodingException;
+import java.math.BigDecimal;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.MediaType;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.person.enums.TermsAndConditions;
+import uk.gov.homeoffice.digital.sas.person.model.Person;
+import uk.gov.homeoffice.digital.sas.person.testutils.KafkaConsumer;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+@AutoConfigureMockMvc
+@ComponentScan("uk.gov.homeoffice.digital.sas.person")
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {
+        "listeners=PLAINTEXT://localhost:3333",
+        "port=3333"
+    }
+)
+class CallistoPersonRestApiIntegrationTest {
+
+  private static final int CONSUMER_TIMEOUT = 10;
+  private static final String TENANT_ID = "b7e813a2-bb28-11ec-8422-0242ac120002";
+  private static final String TENANT_ID_PARAM = "?tenantId="+TENANT_ID;
+  private static final String PERSON_URL = "/resources/persons";
+  private Person person;
+
+  @Value("${projectVersion}")
+  private String version;
+
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Autowired
+  private KafkaConsumer kafkaConsumer;
+
+  @BeforeEach
+  void setup() {
+    person = Person.builder()
```

Comment:
I think we should randomise the persons name so that each test is different. At the moment I'm when the message on the queue is tested I can't tell if it's reading the same message all the time as the person object will always be equal